File: spec/php/OpaqueExternalType02ParentTest.php
<?php
namespace Kaitai\Struct\Tests;

class OpaqueExternalType02ParentTest extends TestCase {
    public function testOpaqueExternalType02Parent() {
        $r = OpaqueExternalType02Parent::fromFile(self::SRC_DIR_PATH . '/term_strz.bin');

        $this->assertSame("foo", $r->parent()->child()->s1());
        $this->assertSame("bar", $r->parent()->child()->s2());
        $this->assertSame("|baz@", $r->parent()->child()->s3()->s3());
    }
}

File: spec/go/spec/opaque_external_type_02_parent_test.go
package spec

import (
	"runtime/debug"
	"os"
	"testing"
	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"
	. "test_formats"
	"github.com/stretchr/testify/assert"
)

func TestOpaqueExternalType02Parent(t *testing.T) {
	defer func() {
		if r := recover(); r != nil {
			debug.PrintStack()
			t.Fatal("unexpected panic:", r)
		}
	}()
	f, err := os.Open("../../src/term_strz.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)
	var r OpaqueExternalType02Parent
	err = r.Read(s, &r, &r)
	if err != nil {
		t.Fatal(err)
	}

	assert.EqualValues(t, "foo", r.Parent.Child.S1)
	assert.EqualValues(t, "bar", r.Parent.Child.S2)
	assert.EqualValues(t, "|baz@", r.Parent.Child.S3.S3)
}
